app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -131,10 +133,9 @@ protected void onCreate(final Bundle savedInstanceState) {
                     + "savedInstanceState = [" + savedInstanceState + "]");
         }
 
-        // enable TLS1.1/1.2 for kitkat devices, to fix download and play for media.ccc.de sources
-        if (Build.VERSION.SDK_INT == Build.VERSION_CODES.KITKAT) {
-            TLSSocketFactoryCompat.setAsDefault();
-        }
+        // use reasonable TLS settings (disable insecure, enable secure options)
+        HttpsURLConnection.setDefaultSSLSocketFactory(
```

Comment:
I think this should be done as early as possible.
I would move it to 
https://github.com/TeamNewPipe/NewPipe/blob/18ce86c2edc6fb9a7ad7bf9df642ba55e0101a63/app/src/main/java/org/schabi/newpipe/App.java#L91

Reply:
I can change it if you would like. But I still can't see its usefulness at all. I've tested the app with those lines completly removed and youtube/mediaCCC.de still worked fine on Android 4.4/7.1.2/9/10. I did not test the other platforms/services.